dlib/test/fft.cpp

```diff
@@ -550,7 +551,140 @@ namespace
         func(123); print_spinner();
         func(131);  print_spinner();
     }
-    
+
+    template<typename R1, typename R2>
+    std::complex<R1> complex_cast(const std::complex<R2> &c) {
+        return { static_cast<R1>(c.real()), static_cast<R1>(c.imag()) };
+    }
+
+    template<typename R>
+    void test_stft()
+    {
+        constexpr R tol = std::is_same<R,float>::value ? 1e-4 : 1e-7;
```

Comment:
Can these tol values be smaller?  The values in `STFT_FFT_128_WLEN_64_TONE_512` are already kinda small in general.  Would be cool if the `tol` was lower.  Like this wouldn't catch if all the really small values in `STFT_FFT_128_WLEN_64_TONE_512` were being matched against similarly really small outputs from the dlib code.

Reply:
Yeah totally, I need to figure out what the smallest values are.